### PR TITLE
[T3A5][W15-A3]Zachary Lim Wei Ling

### PR DESCRIPTION
**Pull request by zachylimwl**

### CODE REVIEW

File: src/seedu/addressbook/commands/CommandResult.java [hunk 1]
```diff
@@ -11,7 +11,7 @@
 public class CommandResult {
 
     /** The feedback message to be shown to the user. Contains a description of the execution result */
-    public final String feedbackToUser;
+    private final String feedbackToUser;
```
Comment (okkhoy) [inline src/seedu/addressbook/commands/CommandResult.java]: once you make this final, you need an accessor method; and you need to update the class/method that is using this variable.

File: doc/UserGuide.md [hunk 2]
```diff
@@ -34,7 +34,7 @@ Format: `help`
  
 ### Adding a person: `add`
 Adds a person to the address book<br>
-Format: `add NAME [p]p/PHONE_NUMBER [p]e/EMAIL [p]a/ADDRESS [t/TAG]...` 
+Format: `add NAME [p]p/PHONE_NUMBER [p]e/EMAIL [p]a/BLOCK, STREET, UNIT, POSTALCODE [t/TAG]...`
```
Comment (okkhoy) [inline doc/UserGuide.md]: good to see you have updated the user guide!

File: src/seedu/addressbook/data/person/Address.java [hunk 3]
```diff
@@ -10,9 +10,12 @@
 
     public static final String EXAMPLE = "123, some street";
     public static final String MESSAGE_ADDRESS_CONSTRAINTS = "Person addresses can be in any format";
-    public static final String ADDRESS_VALIDATION_REGEX = ".+";
 
-    public final String value;
+    public String value;
```
Comment (okkhoy) [inline src/seedu/addressbook/data/person/Address.java]: think, why this (making them `public`) may not be a good idea.

File: src/seedu/addressbook/data/person/Address.java [hunk 4]
```diff
@@ -22,17 +25,30 @@
      */
     public Address(String address, boolean isPrivate) throws IllegalValueException {
         this.isPrivate = isPrivate;
-        if (!isValidAddress(address)) {
+        value = address;
```
Comment (okkhoy) [inline src/seedu/addressbook/data/person/Address.java]: if you retain this, you will be introducing inconsistency due to duplication (please recollect the discussion around this during the tutorial)
the idea is to store the address info in 4 parts and remove this..

File: src/seedu/addressbook/data/person/Address.java [hunk 5]
```diff
@@ -22,17 +25,30 @@
...
             throw new IllegalValueException(MESSAGE_ADDRESS_CONSTRAINTS);
         }
-        this.value = address;
     }
 
+	private void extractAddressComponents(String address) {
+		String[] addressComponents = {"", "", "", ""};
+        String[] inputAddressComponents = address.split(",");
+        for (int i = 0; i < inputAddressComponents.length; i++) {
+        	addressComponents[i] = inputAddressComponents[i].trim();
```
Comment (okkhoy) [inline src/seedu/addressbook/data/person/Address.java]: can you think of a better way to handle this than using a temporary array?

File: src/seedu/addressbook/data/person/Address.java [hunk 6]
```diff
@@ -22,17 +25,30 @@
...
+        }
+        block = new Block(addressComponents[0]);
+        street = new Street(addressComponents[1]);
+        unit = new Unit(addressComponents[2]);
+        postalCode = new PostalCode(addressComponents[3]);
+	}
```
Comment (okkhoy) [inline src/seedu/addressbook/data/person/Address.java]: your indentation is off here.

File: src/seedu/addressbook/data/person/Address.java [hunk 7]
```diff
@@ -22,17 +25,30 @@
...
 
+	private void extractAddressComponents(String address) {
+		String[] addressComponents = {"", "", "", ""};
+        String[] inputAddressComponents = address.split(",");
+        for (int i = 0; i < inputAddressComponents.length; i++) {
+        	addressComponents[i] = inputAddressComponents[i].trim();
+        }
+        block = new Block(addressComponents[0]);
+        street = new Street(addressComponents[1]);
+        unit = new Unit(addressComponents[2]);
```
Comment (okkhoy) [inline src/seedu/addressbook/data/person/Address.java]: try and avoid these magic numbers

File: src/seedu/addressbook/data/person/Block.java [hunk 8]
```diff
@@ -0,0 +1,19 @@
+package seedu.addressbook.data.person;
+
```
Comment (okkhoy) [inline src/seedu/addressbook/data/person/Block.java]: Please include a header comment to indicate what this class/method does?

File: src/seedu/addressbook/data/person/Block.java [hunk 9]
```diff
@@ -0,0 +1,19 @@
...
+public class Block {
+	public static final String BLOCK_VALIDATION_REGEX = "\\d+";
+	public String value;
+	
+	public Block() {
```
Comment (okkhoy) [inline src/seedu/addressbook/data/person/Block.java]: Java provides a default constructor anyway, so you don;t really need to do this.

File: src/seedu/addressbook/data/person/Block.java [hunk 10]
```diff
@@ -0,0 +1,19 @@
...
+		
+	}
+	
+	public Block(String block) {
+		this.value = block;
+	}
+	
+	public boolean isValidBlock() {
+		return value.matches(BLOCK_VALIDATION_REGEX) || value == "";
```
Comment (okkhoy) [inline src/seedu/addressbook/data/person/Block.java]: try to make this explicit by storing it in an aptly named variable?

File: test/expected.txt [hunk 11]
```diff
@@ -85,19 +85,19 @@
 || Enter command: || [Command entered:  add Valid Name p/12345 e/valid@e.mail a/valid, address t/invalid_-[.tag]
 || Tags names should be alphanumeric
 || ===================================================
-|| Enter command: || [Command entered:  add Adam Brown p/111111 e/adam@gmail.com a/111, alpha street]
-|| New person added: Adam Brown Phone: 111111 Email: adam@gmail.com Address: 111, alpha street Tags: 
+|| Enter command: || [Command entered:  add Adam Brown p/111111 e/adam@gmail.com a/111, alpha street, #12-345, 123456]
+|| New person added: Adam Brown Phone: 111111 Email: adam@gmail.com Address: 111, alpha street, #12-345, 123456 Tags:
```
Comment (okkhoy) [inline test/expected.txt]: good to see that you have updated the tests as well..

File: src/seedu/addressbook/storage/jaxb/AdaptedPerson.java [hunk 12]
```diff
@@ -60,7 +60,7 @@ public AdaptedPerson(ReadOnlyPerson source) {
 
         address = new AdaptedContactDetail();
         address.isPrivate = source.getAddress().isPrivate();
-        address.value = source.getAddress().value;
+        address.value = source.getAddress().toString();
```
Comment (okkhoy) [inline src/seedu/addressbook/storage/jaxb/AdaptedPerson.java]: i don't see your update to `toString()` method, hence I am assuming you will return the input address as is in `toString`. if this is the case, you will understand that the processed address stored in 4 objects and retrieved later may have inconsistencies in the format,

**Comment by okkhoy** [comment]
@zachylimwl 
i don't think this one has commits for T3A5, but has some others; hence anyway I have reviewed it, please check the comments, ack and close the PR after reading the comments.

Avoid accumulating unrelated changes into a single PR. 
Tip: If you are currently on `branch1` and want to create a new `branch2`, you need to switch back to `master` branch before creating `branch2`. If not, changes in `branch1` will appear in the PR created from `branch2` later.

have proper gitignore file to ignore the metadata etc.,

File: src/seedu/addressbook/data/person/Block.java (hunk 10 above)
Comment (zachylimwl) [inline src/seedu/addressbook/data/person/Block.java]: @okkhoy Make what explicit?

**Comment by zachylimwl** [comment]
@okkhoy Hmm you mentioned to have a proper gitignore file multiple times already, what do you mean exactly by that? I noticed the metadata changes because I change to my local d drive instead of c so I added it into the gitignore file... is the something else I should add? Please advise :) Thank you! :D

**Comment by okkhoy** [comment]
oh! have I, multiple times!? sorry; may be I did not notice it was your
submission again.. sorry. not my intention to harp on the same thing :)
may be you need to remove the metadata from git tracking totally? do try
that, else drop by some time next week we can try figure out.

**Pull request closed without merging**